### PR TITLE
Sentry integration

### PR DESCRIPTION
**Pull request by GildedHonour**

### CODE REVIEW

**Comment by GildedHonour** [comment]
I've added Sentry to appsync.

In appsync there's 'env/dev.env, that is, it's for 'development' environment. Does 'env/prod.env' file get created on a server automatically? Or there's no necessity in it in the first place because it's serverless and only 'dev.env' should be used?


In 'mail-processing' and 'users' apps there's no 'env/' directories. We need to integrate Sentry for these as well. If I create  'env/dev.env' files in each app, and add config values for Sentry into it, will it be picked up automatically by these apps?

We need Sentry for 'front' as well, right?

**Comment by aldarund** [comment]
yes we need sentry for front.
for users i would say skip for now
for mail processing -> no it wont be picked up automatically. its need to be added into serverless.yaml into environemnt section. see how its done in front.end servrelss for example. And i have created dev.env files for mail processing and users now.
now there two stages of backend can be deployed

**Comment by GildedHonour** [comment]
Updated.  

In `users` and `mail_processing` we have `conf/dev.env`, whereas in appsync it's in a different directory - `env/dev.env`, right?

>> "now there two stages of backend can be deployed" --> is anything required from me?

File: front/nuxt.config.js [hunk 1]
```diff
@@ -154,5 +155,10 @@ module.exports = {
         })
       }
     }
+  },
+
+  sentry: {
+    dsn: process.env.SENTRY_DSN,
+    config: {}, // Additional config
```
Comment (aldarund) [inline front/nuxt.config.js]: configure also uploading of sourcemaps to sentry

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (aldarund) [inline yarn.lock]: not needed

File: front/nuxt.config.js [hunk 3]
```diff
@@ -154,5 +155,10 @@ module.exports = {
         })
       }
     }
+  },
+
+  sentry: {
+    dsn: process.env.SENTRY_DSN,
```
Comment (aldarund) [inline front/nuxt.config.js]: and add env var to disable it for dev

**Comment by GildedHonour** [comment]
@aldarund ok.

Also I sent you  emails, have you seen them? May be in spam.

**Comment by GildedHonour** [comment]
Never mind - I worked with the server instead of the localhost.

File: yarn.lock (hunk 2 above)
Comment (GildedHonour) [reply yarn.lock]: removed

File: front/nuxt.config.js (hunk 1 above)
Comment (GildedHonour) [reply front/nuxt.config.js]: In the documentation of Sentry and internet there's no clear decription of how to use sourcemaps with Nuxtjs or Vue. Do you know how to use it?

Comment (GildedHonour) [reply front/nuxt.config.js]: searching by 'sourcemap' in /front finds results only from 'front/node_modules' . Do we use sourcemaps in the project?

Comment (aldarund) [reply front/nuxt.config.js]: see https://github.com/nuxt-community/sentry-module/blob/7c064df4feefe27ff3c63c1940b36b648959c875/README.md#webpackconfig

source map are generated by nuxt via webpack devtool you need to enable it in nuxt config
https://nuxtjs.org/api/configuration-build/#extend

File: front/nuxt.config.js [hunk 4]
```diff
@@ -143,16 +145,27 @@ module.exports = {
       : undefined,
     cache: true,
     modern: true,
+    sourceMap: true,
+    plugins: [
+      new SentryPlugin({
```
Comment (aldarund) [inline front/nuxt.config.js]: its not needed. its already in the nuxt/sentry module
https://github.com/nuxt-community/sentry-module/blob/8d8c60697f8296ad251241813f81895ec5503c5a/lib/module.js#L182
need to set publishRelease option

Comment (GildedHonour) [reply front/nuxt.config.js]: fixed. 

removed "dsn" because it's set already by default

File: front/package.json [hunk 5]
```diff
@@ -57,6 +58,7 @@
     "serverless-domain-manager": "3.3.0",
     "serverless-offline": "5.12.0",
     "serverless-prune-plugin": "1.4.1",
-    "serverless-s3-sync": "1.9.0"
+    "serverless-s3-sync": "1.9.0",
+    "@sentry/webpack-plugin": "1.10.0"
```
Comment (aldarund) [inline front/package.json]: dont need that as its dep of nuxt/sentry already

File: front/nuxt.config.js [hunk 6]
```diff
@@ -2,6 +2,7 @@ require('dotenv').config()
 // const { VuetifyProgressiveModule } = require('vuetify-loader')
 const path = require('path')
 const resolve = require('path').resolve; // eslint-disable-line
+const SentryPlugin = require("@sentry/webpack-plugin");
```
Comment (aldarund) [inline front/nuxt.config.js]: not used import

**Comment by GildedHonour** [comment]
fixed yesterday

**Comment by aldarund** [comment]
and add updated yarn.lock file too for frontend

**Comment by GildedHonour** [comment]
done

**Comment by aldarund** [comment]
dont see it. you added empty root one. Instead of updated front/yarn.lock

**Comment by GildedHonour** [comment]
fixed

**Pull request merged**